src/glossary.rst

```diff
@@ -43,26 +43,36 @@ Glossary
 
 
    retry
-      Tasks configured to retry on failure will return to the ``waiting`` state
-      with a :term:`clock trigger` to delay the next try.
+   task retry
+   try number
+      Tasks can be configured to retry automatically on failure, one or more
+      times. They return to the ``waiting`` state with a :term:`clock trigger`
+      to delay the retry, and only go to the ``failed`` :term:`state <task
+      state>` once the final try fails.
 
-      Any number of retries, with configurable delays between them, are possible.
-      Task jobs can get their own try number from ``$CYLC_TASK_TRY_NUMBER``.
-
-      If the final try fails, the task goes to the ``failed`` :term:`state
-      <task state>`.
+      The task try number increments with every automatic retry, and is
+      passed to the job environment as ``$CYLC_TASK_TRY_NUMBER``.
 
       .. seealso::
 
          * :ref:`Cylc User Guide <TaskRetries>`
 
 
+   submit number
+   task submit number
+      Every time a task re-runs, whether by automatic :term:`retry` or manual
+      triggering, its submit number increments. Submit number appears in the
+      job log path so that log files don't get overwritten.
```

Comment:
Suggest adding:
It is passed to the job environment as ``$CYLC_TASK_SUBMIT_NUMBER``.